python-package/lightgbm/plotting.py

```diff
@@ -422,15 +422,15 @@ def add(root, total_count, parent=None, decision=None):
             add(root['left_child'], total_count, name, l_dec)
             add(root['right_child'], total_count, name, r_dec)
         else:  # leaf
-            name = 'leaf{0}'.format(root['leaf_index'])
-            label = 'leaf {0}: '.format(root['leaf_index'])
-            label += '<B>{0}</B>'.format(_float2str(root['leaf_value'], precision))
+            name = f"leaf{root['leaf_index']}"
+            label = f"leaf {root['leaf_index']}: "
+            label += f"<B>{_float2str(root['leaf_value'], precision)}</B>"
             if 'leaf_weight' in show_info:
-                label += '<br/>{0} weight'.format(_float2str(root['leaf_weight'], precision))
+                label += f"<br/>{_float2str(root['leaf_weight'], precision)} weight"
             if 'leaf_count' in show_info:
-                label += '<br/>count: {0}'.format(root['leaf_count'])
+                label += f"<br/>count: {root['leaf_count']}"
             if "data_percentage" in show_info:
-                label += '<br/>{0}% of data'.format(_float2str(root['leaf_count'] / total_count * 100, 2))
+                label += f"<br/>{_float2str(root['leaf_count'] / total_count * 100, 2)}% of data"
             label = "<" + label + ">"
```

Comment:
```suggestion
            label = f"<{label}>"
```

can you also please change this line? Concatenation with `+` is another type of string formatting we'd like to replace as part of #4136.

Reply:
And can you please make the same change on line 420? I'm not able to comment on it.

Reply:
Everything using `+=` is fine to leave just as you have, but for the first assignments with just `=`, we'd like to replace those with f-strings.